### ISSUE TITLE
Promising project, now death?

### ISSUE THREAD

**Issue by Jarrodsz** [serenadeai/serenade#30]
This looks promising 
Only with over 1 year of no more commits on this github we can assume this project is death?
Used to build a commercial version available somewhere else?

**Comment by saleh-mir** [comment]
+ 1

I would pay for this to see it continue development. I really need it

**Comment by 0A7V9A** [comment]
i agree. we can fork this project  and continue working while inviting other great developers

**Comment by ui20** [comment]
it is very sad because this is the best solution out there that i'm aware of. i've tried talon but it doesn't really come close. is there a way we can get this thing rolling again?

**Comment by vanadu** [comment]
Someone correct me if I'm wrong, but it looks like Matt Wiehoff scored $2.1M to develop a commercial Serenade product in 2020 then made it open source in 2022,  took a full-time at OpenAI in July 2022 when he stopped work on Serenade. If someone has LinkedIn Premium or Twitter Verified, maybe they can contact him and ask him what his intentions are, or at least whether he can pay for the server and help someone else pick up this ball.

**Comment by bmp713** [comment]
I would definitely pay for Serenade to continue support. Or even better I wish it would overtake Talon for people with RSI. Serenade is far easier to use and customize.

**Comment by vanadu** [comment]
Matt recently posted on Discord about the server issue - my apologies if it seemed I was throwing shade on the investment capital -- not the case. I just wish he'd been able to secure more funding or make it work financially. About Talon - the editing features, particularly selecting, copying, pasting, deleting are far beyond what Serenade was able to accomplish before dev stopped. Serenade just can not do selection effectively nor can you define complex custom voice commands. Having said that, the voice code search and insertion in Serenade are better out of the box than Talon's. But for me, the select functionality is so far superior in Talon, everything else is almost irrelevant. I think you should check that out again -- even if Serenade starts moving again, it's never going to be able to catch up wrt to editing functionality.

**Comment by ui20** [comment]
> Matt recently posted on Discord about the server issue - my apologies if it seemed I was throwing shade on the investment capital -- not the case. I just wish he'd been able to secure more funding or make it work financially. About Talon - the editing features, particularly selecting, copying, pasting, deleting are far beyond what Serenade was able to accomplish before dev stopped. Serenade just can not do selection effectively nor can you define complex custom voice commands. Having said that, the voice code search and insertion in Serenade are better out of the box than Talon's. But for me, the select functionality is so far superior in Talon, everything else is almost irrelevant. I think you should check that out again -- even if Serenade starts moving again, it's never going to be able to catch up wrt to editing functionality.

could you be more specific? you can pretty much select and copy anything you like such as: select block, select line, select third word, copy variable, copy function etc. also navigating the code is very easy compared to talon. I tried to replicate some of the features from serenade in talon with custom commands but it was impossible and I gave up. Oh and being able to change words and characters by just saying change x to y is incredible.

**Comment by jeremiahrose** [comment]
I also like Serenade better than Talon, it's more intuitive for me. And very functional! The main downside was the speech recognition engine consistently misunderstood my Australian accent.

It looks like a fork will be necessary as no PRs have been merged to this repo since July 2022.

Should we put together a small dedicated group to maintain a fork and add local speech recognition capabilities using OpenAI Whisper or similar?

I'm able to get near-to-real time, highly accurate transcriptions using whisper.cpp + VAD running locally on my M1 Mac. Integrating this as an additional engine to Serenade should be achievable right?

**Comment by liam-k** [comment]
Hi, I just discovered this project and I think it's pretty awesome. I'm not very knowledgeable in AI (yet), but I already found a few issues in Serenade I’d like to contribute to which are probably not very hard to fix. For example, lack of support for forks of VSCode or Chromium (which should work out of the box with the current extensions, just a matter of app recognition). I’d also love to see a version with whisper in this.

So I’d be interested in joining a group of maintainers. Is this happening on Discord?

**Comment by katarinaslama** [comment]
Yes! A few of as are discussing contributing/the future of serenade on Discord. See here: https://discord.com/channels/807359313940709436/1192288254167351436

**Comment by jeremiahrose** [comment]
What were the results of the discussion? And will the next meeting be on Jan 23rd at 12pm PST? How do we join - is there a meeting link? Thanks!

**Comment by BFergerson** [comment]
If you're looking for a JetBrains alternative for voice coding, I've been working on the [Voqal Assistant](https://github.com/voqal/voqal). It's not exactly the voice coding that Serenade does, but I would be open to offering more low-level voice coding.

**Comment by MattRey** [comment]
Hi Katarina, I know I'm a few months late to the party but this discord link doesn't seem to work for me. Is there still an active discussion in the future of serenade that you know of and, if so, could you send me an invite? Thanks so much. Matt

**Comment by MattRey** [comment]
Hi @katarinaslama I know I'm a few months late to the party but the discord link doesn't seem to work for me. Is there still an active discussion in the future of serenade that you know of and, if so, could you send me an invite? Thanks so much. Matt

**Comment by atljoseph** [comment]
Seems like nobody ever posted the link to the fork here. That is, if it exists

**Comment by vanderstack** [comment]
Looking at the list of existing forks shown in GitHub it seems no one has a fork with meaningful updates. I’m investigating how I can integrate this into a workflow with GitHub copilot and other LLMs like OpenAI o1 so that I can do meaningful development on mobile devices without a keyboard such as phone/tablet/glasses/vr. Not making any commitments but if I make meaningful progress commits will be pushed to my fork.

**Comment by BFergerson** [comment]
Since I last posted here, there have been significant updates to [Voqal](https://github.com/voqal/voqal). It now works outside JetBrains' IDEs, allowing you to use it for general voice control over your computer. Anyone willing to give feedback is welcome to receive a free lifetime subscription. It is LLM native, so if you are looking to generate code via voice with LLMs like OpenAI/Llama/etc., it has already been implemented.

Here is an example video showing Voqal controlling my browser and Visual Studio Code:

https://github.com/user-attachments/assets/cf748493-0f4c-4ac4-88e5-3a647ac16224

**Comment by vanderstack** [comment]
> Anyone willing to give feedback is welcome to receive a free lifetime subscription. 

This looks like a great solution and I would love to help out in exchange for a lifetime subscription. Any guidance on what kind of feedback you are looking for? Right now this is how I'm interested in using Voqal:

Given this is desktop software my plan right now is to use proxmox to make a windows VM with remote desktop available which voqal will be installed on and set to run on startup. I'll use docker-compose to host a guacamole container which manages the remote connection into the machine in order to expose it to a traditional web browser. Finally I'll use either a tailnet VPN or a cloudflare tunnel to make this guacamole install available to my mobile device while away from my home network. The final result should be that while on the go I can orchestrate Voqal.

Once I have the full stack working I would be happy to make some content about using voqal from my iphone/ipad with the mic in my earbuds and my xreal AR glasses for a large display. For me this means I can interact with AI to get things done do while playing fetch with my dog in the backyard, although I expect this would be valuable to a broader audience such as commuters working from a train or airplane.

If this sounds like a compelling use case you would like to be able to showcase or collaborate on let me know what platform I can reach you privately on so I'm not spamming this thread.

**Comment by Splines** [comment]
The last commit on the master branch dates back 3 years ago. It'd be awesome if some of the main developers could give a short update on this project's state and maybe add a warning banner in the Readme indicating that this repo is no longer actively maintained. Downloading the `.exe` from the server also takes a very long time (my browser indicates 7h for 750MB on a reasonably fast Internet connection).

But thanks for having tried out this idea, which looks very promising. I hope it will inspire future developments/forks or even new products. Speaking of that, does anybody know a good voice app (for Windows) with the following requirements?

- The app should allow me to naturally interact with VSCode, e.g. by saying "Open file `edit.html` in the `questionnaires/edit` folder", "Open terminal" or "Search in whole codebase for the string `data-bs-content`".
- The app should allow me to "normally" code without a code AI in Ruby on Rails, e.g. I want to say something like this (as seen on the [Serenade website](https://serenade.ai/))
```js
// "Next line, add db dot query comma brackets id"
=> db.run(query, [id]);
```

Or something like:

```rb
# "Return render turbo_stream refresh action and show alert 'vignettes Not editable'"
return render turbo_stream: turbo_stream.action(:refresh, ""), alert: t("vignettes.not_editable")
```

- The app should also allow me to integrate with GitHub copilot, i.e. it lets me say something like

```
Copilot, Refactor the function `abc` by extracting a method out of it.
(Copilot could be some kind of clue such that the app knows to use Copilot,
e.g. by simply pasting my prompt into the VSCode Copilot chat pane.
```

Ideally, I wouldn't have to learn any voice commands. Instead, the voice AI itself would be an LLM that tries to understand what I want to do, then performs the corresponding action.
